express/parsers/structure.py

```diff
@@ -41,6 +68,89 @@ def __init__(self, *args, **kwargs):
         self.lattice_only_structure = mg.Structure.from_str(self.structure_string, self.structure_format)  # deepcopy
         self.lattice_only_structure.remove_sites(range(1, len(self.structure.sites)))
 
+        # testing
+
+    def get_inchi_run(self):
+        return inchi_run
+
+    def get_inchi_null(self):
+        """
+        Function returns "Not Available" value for InChI when it cannot be calculated.
+
+        Returns:
+            Str
+        """
+        inchi_key_str = {
+            'name': 'inchi',
+            'value': ''
+        }
+        return inchi_key_str
+
+    def create_pybel_smi_from_poscar(self):
+        """
+        Function using ase to convert the POSCAR formatted string of a structure
+        into an XYZ formatted text file for that structure.
+
+        Then pybel converts the XYZ formatted text file into a SMILES format.
+
+        Returns:
+            Str
+        """
+        fi = "geom.xyz"
+        f = open(fi, "w")
+        os.chmod(fi, 0o777)
+        fstr = StringIO(self.structure_string)
+        ase_pos = ase.io.read(fstr, format="vasp")
+        xyz_file = ase.io.write(fi, ase_pos, format='xyz')
+        psmi = list(pybel.readfile('xyz', 'geom.xyz'))[0]
+        f.close()
+        return psmi
+
+    def get_inchi(self):
+        """
+        Function calculates the International Chemical Identifier (InChI) string for a given structure.
+
+        Returns:
+            Str
+        """
+        psmi = self.create_pybel_smi_from_poscar()
+        self.inchi = psmi.write("inchi")
+        inchi_short = self.inchi.split("=")
+        inchi_short = inchi_short[1]
+        inchi_str = {
+            "name": "inchi",
+            "value": inchi_short
+        }
```

Comment:
The initial `inchi` format from `pybel` looks like `InChI=1S/C6H6/c1-2-4-6-5-3-1/h1-6H` so I format it to not have the `InChI=` for display on the webapp.